File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int> mpp;
        for(int i=0;i<n;i++)
        {
            int num1=nums[i];
            int num2=target-num1;
            if(mpp.find(num2)!=mpp.end())
            {
                return {i,mpp[num2]};
            }
            mpp[num1]=i;
        }
        return vector<int> {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        Map<Integer,Integer> mpp=new HashMap<>();
        for(int i=0;i<n;i++)
        {
            int num1=nums[i];
            int num2=target-num1;

            if(mpp.containsKey(num2))
            return new int[] {i,mpp.get(num2)};

            mpp.put(num1,i);
        }

        return new int[] {-1,-1};
    }
}